botorch/generation/gen.py

```diff
@@ -98,6 +99,10 @@ def gen_candidates_scipy(
     )
 
     def f(x):
+        if np.isnan(x).any():
+            raise RuntimeError(
+                f"{np.isnan(x).sum()} elements of the {x.shape} tensor `x` are NaN."
```

Comment:
```suggestion
                f"{np.isnan(x).sum()} elements of the {x.size} array `x` are NaN."
```

This array is 1-d and kind of hard to interpret, so I think it is cleaner to just say x/N elements are NaN